src/modules/users/users.service.ts

```diff
@@ -1,28 +1,129 @@
-import { Injectable } from '@nestjs/common';
+import { Wishlist } from '#/modules/wishlists/entities/wishlist.entity';
+import { UpdateProfileDto } from './dto/requests/update-profile.dto';
+import {
+  ForbiddenException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Account } from './entities/account.entity';
+import { Profile } from './entities/profile.entity';
+import { Repository } from 'typeorm';
+import { Roles } from '#/shared/constants/user-roles.constant';
 
 @Injectable()
 export class UserService {
-  constructor(@InjectRepository(Account) private userRepo: Account) {}
+  constructor(
+    @InjectRepository(Account)
+    private userAccountRepository: Repository<Account>,
+    @InjectRepository(Profile)
+    private userProfileRepository: Repository<Profile>,
+    @InjectRepository(Wishlist)
+    private wishlistRepository: Repository<Wishlist>,
+  ) {}
 
-  create() {
-    return 'This action adds a new user';
+  async listUsers() {
+    const users = await this.userAccountRepository.find({
+      select: ['id', 'email', 'isDeleted', 'role', 'createdAt', 'updatedAt'],
+    });
+
+    return users;
   }
 
-  findAll() {
-    return `This action returns all user`;
+  async getProfile(accountId: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+      relations: ['profile'],
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
+
+    return account.profile;
+  }
+
+  async getUserWishlist(accountId: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+    });
+    if (!account) {
+      throw new NotFoundException('Account not found');
+    }
+
+    const userWishlist = await this.wishlistRepository.find({
+      where: {
+        accountId,
+      },
+    });
+
+    return userWishlist;
+  }
+
+  async updateProfile(accountId: string, updateProfileDto: UpdateProfileDto) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+      relations: ['profile'],
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
```

Comment:
No need to search for an account here. Just search profile by accountID since the account is already validated by guard. 
Better just create a profile with empty values or values from incoming updateDto for an account if the profile doesn't exist, because its absence is usually not the users' fault. There's nothing a user can do if they have an account but not a profile. Error doesn't make any sense.

Reply:
Fixed